pkg/istio/authorizationpolicy/controller.go

```diff
@@ -369,3 +385,43 @@ func (c *Controller) checkAuthzEnabledAnnotation(serviceObj *corev1.Service) boo
 	}
 	return false
 }
+
+// cleanUpStaleAP deletes the existing Authorization Policy associated to the service which is switching back from
+// Authorization Policy Enabled back to SR/SRB
+func (c *Controller) cleanUpStaleAP() error {
+	// Fetch the Authorization Policies present across all the namespaces
+	currentAPList, err := c.configStoreCache.List(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), "")
+	if err != nil {
+		return fmt.Errorf("Error while fetching the Authorization Policy resources from cache : %v", err.Error())
+	}
+
+	for _, currAP := range currentAPList {
+		nsSvcArr := strings.Split(currAP.Key(), "/")
+		serviceName := nsSvcArr[2]
+		serviceNamespace := nsSvcArr[1]
+
+		// Creating the Item to pass to the apiHandler
+		// with a delete event
+		key := serviceNamespace + "/" + serviceName
+		cbHandler := c.getCallbackHandler(key)
+
+		item := &common.Item{
+			Operation:       model.EventDelete,
+			Resource:        currAP,
+			CallbackHandler: cbHandler,
+		}
+
+		eHandler := &c.apiHandler
+
+		// Check if the Authorization Policy is enabled for the service through ap-enabled-list
+		if !c.componentEnabledAuthzPolicy.IsEnabled(serviceName, serviceNamespace) {
+			log.Infof("Deleting stale AP, namespace: %v service name: %v", serviceNamespace, serviceName)
```

Comment:
Before deleting, need to check if ap has override annotation set, we don't want to delete user's manually applied ap.